tt_eager/tt_dnn/op_library/transformer_tms/transformer_tms.cpp

```diff
@@ -663,6 +663,59 @@ tt::stl::reflection::Attributes SSM1DSumReduce::attributes() const {
     };
 }
 
+void SSMPrefixScan::validate(const std::vector<Tensor>& input_tensors) const {
+    TT_FATAL(input_tensors.size() == 2, "Expected 2 input tensors");
+
+    const auto& a = input_tensors.at(0);
+    const auto& bx = input_tensors.at(1);
+
+    TT_FATAL(a.dtype() == bx.dtype(), "Expected input tensors to have the same data type");
+
+    TT_FATAL(a.layout() == Layout::TILE && bx.layout() == Layout::TILE, "Expected input tensors to be tile layout");
+
+    TT_FATAL(a.get_shape() == bx.shape(), "Expected input tensors to have the same shape");
+
+    const auto& shape = a.get_shape();
+    TT_FATAL(shape.rank() == 4, "Expected input tensors to be rank 4");
+    TT_FATAL(shape[0] == 1 && shape[1] == 1, "Dimension 0 and 1 should be size 1");
+    TT_FATAL(shape[2] >= TILE_HEIGHT && shape[2] % TILE_HEIGHT == 0, "Sequence length should be a multiple of 32");
+     // TT_FATAL(shape[3] == HIDDEN_SIZE * LATENT_SIZE, "Hidden state must be shape 2EN (E=2560, N=32)");
+
+    TT_FATAL(a.is_sharded() && bx.is_sharded(), "Expected input tensors to be sharded");
+    TT_FATAL(
+        a.shard_spec().value().orientation == ShardOrientation::ROW_MAJOR,
+        "Expected A tensor to be row major orientation");
+    TT_FATAL(
+        bx.shard_spec().value().orientation == ShardOrientation::ROW_MAJOR,
+        "Expected Bx tensor to be row major orientation");
+}
+
+std::vector<Shape> SSMPrefixScan::compute_output_shapes(const std::vector<Tensor>& input_tensors) const {
+    const auto& a = input_tensors.at(0);
+    return {a.get_legacy_shape()};
```

Comment:
You use `get_shape` in your validate and `get_legacy_shape` here. Since these could represent different things, it's better to be consistent.

Reply:
Fixed